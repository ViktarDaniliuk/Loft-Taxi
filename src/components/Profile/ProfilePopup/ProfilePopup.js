import React, { Component } from 'react';
import ProfilePopupMod from './ProfilePopup.module.css';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

class ProfilePopup extends Component {
   state = {
      cardNumber: "",
      validity: "",
      userName: "",
      CVCcode: ""
   };

   static propTypes = {
      handleChangePaymentData: PropTypes.func
   };

   handleSubmit = e => {
      e.preventDefault();


   };

   handleCardNumberChange = e => {
      this.setState({ cardNumber: e.target.value });
   };

   handleValidityChange = e => {
      this.setState({ validity: e.target.value });
   };

   handleUserNameChange = e => {
      this.setState({ userName: e.target.value });
   };

   handleCVCcodeChange = e => {
      this.setState({ CVCcode: e.target.value });
   };

   render () {
      // console.log('ProfilePopup props: ', this.props);
      return (
         <div className={ ProfilePopupMod.profile_popup }>
            <h1>Профиль</h1>
            <p>Способ оплаты</p>
            <form onSubmit={this.handleSubmit}>
               <div>
                  <div>
                     <label>
                        Номер карты:
                        <input type="text" value={this.state.cardNumber} onChange={this.handleCardNumberChange} placeholder="1234 5678 1234 5678" />
                     </label>
                     <label>
                        Срок действия:
                        <input type="text" value={this.state.validity} onChange={this.handleValidityChange} placeholder="00/00" />
                     </label>
                  </div>
                  <div>
                     <label>
                        Имя владельца:
                        <input type="text" value={this.state.userName} onChange={this.handleUserNameChange} placeholder="USER NAME" />
                     </label>
                     <label>
                        CVC:
                        <input type="password" value={this.state.CVCcode} onChange={this.handleCVCcodeChange} placeholder="***" />
                     </label>
                  </div>
               </div>
               <Link to="/map">
                  <input type="submit" value="Сохранить" onClick={ () => { this.props.handleChangePaymentData();
                  }} />
               </Link>
            </form>
         </div>
      );
   }
}

export default ProfilePopup;