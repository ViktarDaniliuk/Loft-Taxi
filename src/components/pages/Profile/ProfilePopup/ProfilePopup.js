import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Form, Field } from 'react-final-form';
import ProfilePopupMod from './ProfilePopup.module.css';
import { sendPaymentDataRequest } from '../../../../redux/actions';
import CardNumberInput from './fields/CardNumberInput';
import ValidityInput from './fields/ValidityInput';
import UserCVCcodeInput from './fields/UserCVCcode';
import { history } from '../../../../history';

export class ProfilePopup extends Component {
   state = {
      sendedData: false,
      disabled: "disabled"
   };

   static propTypes = {
      sendPaymentDataRequest: PropTypes.func.isRequired,
      cardNumber: PropTypes.string,
      validity: PropTypes.string,
      userFullName: PropTypes.string,
      CVCcode: PropTypes.string
   };

   onChangePaymentData = (values) => {

      const { sendPaymentDataRequest } = this.props;
      const { cardNumber, validity, userFullName, CVCcode } = values;

      if (!values.cardNumber || !values.validity || !values.userFullName || !values.CVCcode) return;

      sendPaymentDataRequest(cardNumber, validity, userFullName, CVCcode);
      this.setState({
         sendedData: true
      });
   };

   onChangePage = () => {
      history.push('/map');
   };

   render () {
      console.log('ProfilePopup rendered: ', new Date().getHours(), ':', new Date().getMinutes(), ':', new Date().getSeconds(), ':', new Date().getMilliseconds());
      return (
         <div className={ ProfilePopupMod.profile_popup }>
            <h1>Профиль</h1>
            { this.props.isPaymentData === true && 
            this.state.sendedData === true && 
            <>
               <p>Платежные данные обновлены. Теперь вы можете заказывать такси</p>
               <button type="submit" onClick={ this.onChangePage }>
                  Перейти на карту
               </button>
            </> }
            { this.state.sendedData === false &&
            <>
               <p>Способ оплаты</p>
               <Form onSubmit= { this.onChangePaymentData }>
                  {({ handleSubmit }) => (
                     <form onSubmit={ handleSubmit }>
                        <div>
                           <div>
                              <label>
                                 Номер карты:
                                 <Field 
                                    name="cardNumber" 
                                    initialValue={ this.props.cardNumber }
                                    component={ CardNumberInput } 
                                 />
                              </label>
                              <label>
                                 Срок действия:
                                 <Field 
                                    name="validity" 
                                    initialValue={ this.props.validity }
                                    component={ ValidityInput } 
                                 />
                              </label>
                           </div>
                           <div>
                              <label>
                                 Имя владельца:
                                 <Field 
                                    name="userFullName" 
                                    initialValue={ this.props.userFullName }
                                    component="input"
                                    placeholder="FULL USER NAME" 
                                 />
                              </label>
                              <label>
                                 CVC:
                                 <Field 
                                    name="CVCcode"
                                    type="password"
                                    initialValue={ this.props.CVCcode }
                                    component={ UserCVCcodeInput }
                                 />
                              </label>
                           </div>
                        </div>
                        <button type="submit">
                           Сохранить
                        </button>
                     </form>
                  )}
               </Form>
            </>}
         </div>
      );
   }
};

const mapStateToProps = state => {
   return {
      cardNumber: state.cardData.cardNumber,
      validity: state.cardData.validity,
      userFullName: state.cardData.userFullName,
      CVCcode: state.cardData.CVCcode,
      isPaymentData: state.cardData.isPaymentData
   };
};

const mapDispatchToProps = dispatch => {
   return {
      sendPaymentDataRequest: (cardNumber, validity, userFullName, CVCcode) => {
         dispatch(sendPaymentDataRequest(cardNumber, validity, userFullName, CVCcode));
      }
   };
};

export const WrappedProfilePopup = connect(mapStateToProps, mapDispatchToProps)(ProfilePopup);