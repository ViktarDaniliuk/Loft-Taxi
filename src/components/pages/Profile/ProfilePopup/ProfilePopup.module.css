.profile_popup {
   width: 800px;
   text-align: center;
   background-color: #fff;
   border-radius: 5px;
   padding: 35px;
}

.profile_popup h1 {
   margin-top: 0;
}

.profile_popup button[type="submit"] {
   margin: 35px auto 0;
}

.profile_popup form > div {
   display: flex;
   justify-content: space-between;
}

.profile_popup form > div > div {
   width: 320px;
   height: 170px;
   border-radius: 5px;
   box-shadow: 0px 2px 7px -3px #000000;
   padding: 30px;
   text-align: left;
}

.profile_popup form > div > div input {
   height: 40px;
   font-size: 22px;
   padding-left: 5px;
}

.profile_popup form > div label:nth-of-type(1) input {
   width: auto;
   margin-right: 0;
}

.profile_popup form > div > div label:nth-of-type(2) input {
   width: 130px;
}

.profile_popup form > div label:nth-of-type(1) {
   margin-top: 0;
}

.profile_popup form > div label:nth-of-type(2) {
   margin-top: 30px;
}