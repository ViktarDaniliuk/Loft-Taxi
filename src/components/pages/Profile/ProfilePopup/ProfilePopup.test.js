import React from 'react';
import ReactDOM from 'react-dom';
import { ProfilePopup } from './ProfilePopup';

it('render  without crashing', () => {
   const div = document.createElement('div');
   ReactDOM.render(<ProfilePopup />, div);
   ReactDOM.unmountComponentAtNode(div);
});