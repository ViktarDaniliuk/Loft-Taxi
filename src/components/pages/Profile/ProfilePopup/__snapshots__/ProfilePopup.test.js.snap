// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePopup Snapshot: 
    <div
      className="profile_popup"
    >
      <h1>
        Профиль
      </h1>
      <p>
        Способ оплаты
      </p>
      <form
        onSubmit={[Function]}
      >
        <div>
          <div>
            <label>
              Номер карты:
              <input
                onChange={[Function]}
                placeholder="1234 5678 1234 5678"
                type="text"
                value=""
              />
            </label>
            <label>
              Срок действия:
              <input
                onChange={[Function]}
                placeholder="00/00"
                type="text"
                value=""
              />
            </label>
          </div>
          <div>
            <label>
               Имя владельца:
               <input
                  onChange={[Function]}
                  placeholder="USER NAME"
                  type="text"
                  value=""
               />
            </label>
            <label>
               CVC:
               <input
                  onChange={[Function]}
                  placeholder="***"
                  type="password"
                  value=""
               />
            </label>
            </div>
         </div>
         <input
            onClick={[Function]}
            type="submit"
            value="Сохранить"
         />
         </form>
      </div>
    1`] = `
<div
  className="profile_popup"
>
  <h1>
    Профиль
  </h1>
  <p>
    Способ оплаты
  </p>
  <form
    onSubmit={[Function]}
  >
    <div>
      <div>
        <label>
          Номер карты:
          <input
            onChange={[Function]}
            placeholder="1234 5678 1234 5678"
            type="text"
            value=""
          />
        </label>
        <label>
          Срок действия:
          <input
            onChange={[Function]}
            placeholder="00/00"
            type="text"
            value=""
          />
        </label>
      </div>
      <div>
        <label>
          Имя владельца:
          <input
            onChange={[Function]}
            placeholder="USER NAME"
            type="text"
            value=""
          />
        </label>
        <label>
          CVC:
          <input
            onChange={[Function]}
            placeholder="***"
            type="password"
            value=""
          />
        </label>
      </div>
    </div>
    <input
      onClick={[Function]}
      type="submit"
      value="Сохранить"
    />
  </form>
</div>
`;

exports[`ProfilePopup Snapshot: 
    <div
      className="profile_popup"
    >
      <h1>
        Профиль
      </h1>
      <p>
        Способ оплаты
      </p>
      <form
        onSubmit={[Function]}
      >
        <div>
          <div>
            <label>
              Номер карты:
              <input
                onChange={[Function]}
                placeholder="1234 5678 1234 5678"
                type="text"
                value=""
              />
            </label>
            <label>
              Срок действия:
              <input
                onChange={[Function]}
                placeholder="00/00"
                type="text"
                value=""
              />
            </label>
          </div>
          <div>
            <label>
              Имя владельца:
              <input
                onChange={[Function]}
                placeholder="USER NAME"
                type="text"
                value=""
              />
            </label>
            <label>
               CVC:
               <input
                  onChange={[Function]}
                  placeholder="***"
                  type="password"
                  value=""
               />
               </label>
            </div>
         </div>
         <input
            onClick={[Function]}
            type="submit"
            value="Сохранить"
         />
         </form>
      </div>
    1`] = `
<div
  className="profile_popup"
>
  <h1>
    Профиль
  </h1>
  <p>
    Способ оплаты
  </p>
  <form
    onSubmit={[Function]}
  >
    <div>
      <div>
        <label>
          Номер карты:
          <input
            onChange={[Function]}
            placeholder="1234 5678 1234 5678"
            type="text"
            value=""
          />
        </label>
        <label>
          Срок действия:
          <input
            onChange={[Function]}
            placeholder="00/00"
            type="text"
            value=""
          />
        </label>
      </div>
      <div>
        <label>
          Имя владельца:
          <input
            onChange={[Function]}
            placeholder="USER NAME"
            type="text"
            value=""
          />
        </label>
        <label>
          CVC:
          <input
            onChange={[Function]}
            placeholder="***"
            type="password"
            value=""
          />
        </label>
      </div>
    </div>
    <input
      onClick={[Function]}
      type="submit"
      value="Сохранить"
    />
  </form>
</div>
`;

exports[`ProfilePopup Snapshot: 
    <div
      className="profile_popup"
    >
      <h1>
        Профиль
      </h1>
      <p>
        Способ оплаты
      </p>
      <form
        onSubmit={[Function]}
      >
        <div>
          <div>
            <label>
              Номер карты:
              <input
                onChange={[Function]}
                placeholder="1234 5678 1234 5678"
                type="text"
                value=""
              />
            </label>
            <label>
              Срок действия:
              <input
                onChange={[Function]}
                placeholder="00/00"
                type="text"
                value=""
              />
            </label>
          </div>
          <div>
            <label>
              Имя владельца:
              <input
                onChange={[Function]}
                placeholder="USER NAME"
                type="text"
                value=""
              />
            </label>
            <label>
               CVC:
               <input
                  onChange={[Function]}
                  placeholder="***"
                  type="password"
                  value=""
               />
            </label>
            </div>
         </div>
         <input
            onClick={[Function]}
            type="submit"
            value="Сохранить"
         />
         </form>
      </div>
    1`] = `
<div
  className="profile_popup"
>
  <h1>
    Профиль
  </h1>
  <p>
    Способ оплаты
  </p>
  <form
    onSubmit={[Function]}
  >
    <div>
      <div>
        <label>
          Номер карты:
          <input
            onChange={[Function]}
            placeholder="1234 5678 1234 5678"
            type="text"
            value=""
          />
        </label>
        <label>
          Срок действия:
          <input
            onChange={[Function]}
            placeholder="00/00"
            type="text"
            value=""
          />
        </label>
      </div>
      <div>
        <label>
          Имя владельца:
          <input
            onChange={[Function]}
            placeholder="USER NAME"
            type="text"
            value=""
          />
        </label>
        <label>
          CVC:
          <input
            onChange={[Function]}
            placeholder="***"
            type="password"
            value=""
          />
        </label>
      </div>
    </div>
    <input
      onClick={[Function]}
      type="submit"
      value="Сохранить"
    />
  </form>
</div>
`;

exports[`ProfilePopup Snapshot: 
    <div
      className="profile_popup"
    >
      <h1>
        Профиль
      </h1>
      <p>
        Способ оплаты
      </p>
      <form
        onSubmit={[Function]}
      >
        <div>
          <div>
            <label>
              Номер карты:
              <input
                onChange={[Function]}
                placeholder="1234 5678 1234 5678"
                type="text"
                value=""
              />
            </label>
            <label>
              Срок действия:
              <input
                onChange={[Function]}
                placeholder="00/00"
                type="text"
                value=""
              />
            </label>
          </div>
          <div>
            <label>
              Имя владельца:
              <input
                onChange={[Function]}
                placeholder="USER NAME"
                type="text"
                value=""
              />
            </label>
            <label>
              CVC:
              <input
                onChange={[Function]}
                placeholder="***"
                type="password"
                value=""
              />
            </label>
          </div>
        </div>
        <input
          onClick={[Function]}
          type="submit"
          value="Сохранить"
        />
      </form>
    </div>
   1`] = `
<div
  className="profile_popup"
>
  <h1>
    Профиль
  </h1>
  <p>
    Способ оплаты
  </p>
  <form
    onSubmit={[Function]}
  >
    <div>
      <div>
        <label>
          Номер карты:
          <input
            onChange={[Function]}
            placeholder="1234 5678 1234 5678"
            type="text"
            value=""
          />
        </label>
        <label>
          Срок действия:
          <input
            onChange={[Function]}
            placeholder="00/00"
            type="text"
            value=""
          />
        </label>
      </div>
      <div>
        <label>
          Имя владельца:
          <input
            onChange={[Function]}
            placeholder="USER NAME"
            type="text"
            value=""
          />
        </label>
        <label>
          CVC:
          <input
            onChange={[Function]}
            placeholder="***"
            type="password"
            value=""
          />
        </label>
      </div>
    </div>
    <input
      onClick={[Function]}
      type="submit"
      value="Сохранить"
    />
  </form>
</div>
`;
